langchain/src/agents/xml/output_parser.ts

```diff
@@ -38,13 +38,15 @@ export class XMLAgentOutputParser extends AgentActionOutputParser {
    */
   async parse(text: string): Promise<AgentAction | AgentFinish> {
     if (text.includes("</tool>")) {
-      const [tool, toolInput] = text.split("</tool>");
-      const _tool = tool.split("<tool>")[1];
-      const _toolInput = toolInput.split("<tool_input>")[1];
-      return { tool: _tool, toolInput: _toolInput, log: text };
+        const _toolMatch = text.match(/<tool>([^<]*)<\/tool>/);
+        const _tool = _toolMatch ? _toolMatch[1]: '';
+        const _toolInputMatch = text.match(/<tool_input>([^<]*)<\/tool_input>/);
```

Comment:
Should make the closing `<\/tool_input>` optional? I have definitely seen generations where the final stop sequence gets omitted.